deployments/charts/kuma/values.yaml

```diff
@@ -136,18 +184,46 @@ controlPlane:
       clientCertsSecretName: ""
     kdsGlobalServer:
       # -- Secret that contains tls.crt, tls.key for protecting cross cluster communication
+      # If kdsGlobalServerTLSSecret is enabled, the generated secret's name
+      # will be used by default, and this would override it.
       secretName: ""
     kdsZoneClient:
       # -- Secret that contains ca.crt which was used to sign KDS Global server. Used for CP verification
+      # If kdsZoneClientTLSSecret is enabled, the generated secret's name will
+      # be used by default, and this would override it.
       secretName: ""
 
+  kdsGlobalServerTLSSecret:
```

Comment:
Any reason not to have these 2 in tls.kdsGlobalServer?

You could then have the name always be configurable (makes little sense but why not ;)) and then have:

```yaml
kdsZoneClient:
  secretName: ""
  cert: ""
  key: ""
```

You'd have to validate that if cert is set key must be as well. 
if you want to avoid confusion you can always add `provided: false` and add the asserts to make sure people set the things correctly.

Reply:
Yeah I'm fine with that. I think I separated them out because it would give you more options for maintaining backwards compatibility if you change it later.

Reply:
Note that you do still need a "enabled" attribute as well, to maintain compatibility, since the cert validation was optional.